packages/flutter/lib/src/widgets/router.dart

```diff
@@ -967,6 +967,10 @@ class ChildBackButtonDispatcher extends BackButtonDispatcher {
 
   @override
   void takePriority() {
+    if (parent is ChildBackButtonDispatcher) {
```

Comment:
Instead of checking the type here and type casting, could this logic just be implemented in `ChildBackButtonDispatcher.deferTo`, which we would call in line 974 anyways?